diff -ENwbur a/agent/mibgroup/etherlike-mib/data_access/dot3stats_linux.c b/agent/mibgroup/etherlike-mib/data_access/dot3stats_linux.c
--- a/agent/mibgroup/etherlike-mib/data_access/dot3stats_linux.c	2019-05-26 07:34:29.861529615 -0300
+++ b/agent/mibgroup/etherlike-mib/data_access/dot3stats_linux.c	2019-05-26 07:34:44.985306314 -0300
@@ -526,7 +526,7 @@
     u_long value;
     dot3StatsTable_data *data = &rowreq_ctx->data;
     FILE *dev;
-    const char NETDEV_FILE[] = "/proc/net/dev";
+    const char NETDEV_FILE[] = "/tmp/proc/net/dev";
 
     if (_dot3Stats_netlink_get_errorcntrs(rowreq_ctx, name) == 0)
     {
diff -ENwbur a/agent/mibgroup/if-mib/data_access/interface_linux.c b/agent/mibgroup/if-mib/data_access/interface_linux.c
--- a/agent/mibgroup/if-mib/data_access/interface_linux.c	2019-05-26 07:34:30.405519208 -0300
+++ b/agent/mibgroup/if-mib/data_access/interface_linux.c	2019-05-26 07:34:45.549301633 -0300
@@ -591,7 +591,7 @@
  *
  * @retval  0 success
  * @retval -1 no container specified
- * @retval -2 could not open /proc/net/dev
+ * @retval -2 could not open /tmp/proc/net/dev
  * @retval -3 could not create entry (probably malloc)
  */
 int
@@ -615,10 +615,10 @@
         return -1;
     }
 
-    if (!(devin = fopen("/proc/net/dev", "r"))) {
+    if (!(devin = fopen("/tmp/proc/net/dev", "r"))) {
         DEBUGMSGTL(("access:interface",
                     "Failed to load Interface Table (linux1)\n"));
-        snmp_log_perror("interface_linux: cannot open /proc/net/dev");
+        snmp_log_perror("interface_linux: cannot open /tmp/proc/net/dev");
         return -2;
     }
 
@@ -655,11 +655,11 @@
         if (strstr(line, "compressed")) {
             scan_expected = 10;
             DEBUGMSGTL(("access:interface",
-                        "using linux 2.2 kernel /proc/net/dev\n"));
+                        "using linux 2.2 kernel /tmp/proc/net/dev\n"));
         } else {
             scan_expected = 5;
             DEBUGMSGTL(("access:interface",
-                        "using linux 2.0 kernel /proc/net/dev\n"));
+                        "using linux 2.0 kernel /tmp/proc/net/dev\n"));
         }
     }
 
diff -ENwbur a/agent/mibgroup/ip-mib/data_access/ipaddress_ioctl.c b/agent/mibgroup/ip-mib/data_access/ipaddress_ioctl.c
--- a/agent/mibgroup/ip-mib/data_access/ipaddress_ioctl.c	2019-05-26 07:34:30.741512780 -0300
+++ b/agent/mibgroup/ip-mib/data_access/ipaddress_ioctl.c	2019-05-26 07:34:45.893298778 -0300
@@ -655,6 +655,8 @@
 int
 netsnmp_access_ipaddress_ioctl_get_interface_count(int sd, struct ifconf * ifc)
 {
+	return -1;
+	
     int lastlen = 0, i;
     struct ifconf ifc_tmp;
 
diff -ENwbur a/agent/mibgroup/ip-mib/data_access/ipaddress_linux.c b/agent/mibgroup/ip-mib/data_access/ipaddress_linux.c
--- a/agent/mibgroup/ip-mib/data_access/ipaddress_linux.c	2019-05-26 07:34:30.749512627 -0300
+++ b/agent/mibgroup/ip-mib/data_access/ipaddress_linux.c	2019-05-26 07:34:45.901298712 -0300
@@ -232,7 +232,7 @@
     
     netsnmp_assert(NULL != container);
 
-#define PROCFILE "/proc/net/if_inet6"
+#define PROCFILE "/tmp/proc/net/if_inet6"
     if (!(in = fopen(PROCFILE, "r"))) {
         snmp_log_perror("ipaddress_linux: could not open " PROCFILE);
         return -2;
diff -ENwbur a/agent/mibgroup/ip-mib/data_access/ipv6scopezone_linux.c b/agent/mibgroup/ip-mib/data_access/ipv6scopezone_linux.c
--- a/agent/mibgroup/ip-mib/data_access/ipv6scopezone_linux.c	2019-05-26 07:34:30.777512092 -0300
+++ b/agent/mibgroup/ip-mib/data_access/ipv6scopezone_linux.c	2019-05-26 07:34:45.929298479 -0300
@@ -66,7 +66,7 @@
     
     netsnmp_assert(NULL != container);
 
-#define PROCFILE "/proc/net/if_inet6"
+#define PROCFILE "/tmp/proc/net/if_inet6"
     if (!(in = fopen(PROCFILE, "r"))) {
         DEBUGMSGTL(("access:scopezone:container","could not open " PROCFILE "\n"));
         return -2;
diff -ENwbur a/agent/mibgroup/mibII/interfaces.c b/agent/mibgroup/mibII/interfaces.c
--- a/agent/mibgroup/mibII/interfaces.c	2019-05-26 07:34:24.585630573 -0300
+++ b/agent/mibgroup/mibII/interfaces.c	2019-05-26 07:34:40.229345546 -0300
@@ -1557,9 +1557,9 @@
     /*
      * at least linux v1.3.53 says EMFILE without reason... 
      */
-    if (!(devin = fopen("/proc/net/dev", "r"))) {
+    if (!(devin = fopen("/tmp/proc/net/dev", "r"))) {
         close(fd);
-        NETSNMP_LOGONCE((LOG_ERR, "cannot open /proc/net/dev.\n"));
+        NETSNMP_LOGONCE((LOG_ERR, "cannot open /tmp/proc/net/dev.\n"));
         return; /** exit (1); **/
     }
 
@@ -1576,11 +1576,11 @@
     if (strstr(line, "compressed")) {
         scan_line_to_use = scan_line_2_2;
         DEBUGMSGTL(("mibII/interfaces",
-                    "using linux 2.2 kernel /proc/net/dev\n"));
+                    "using linux 2.2 kernel /tmp/proc/net/dev\n"));
     } else {
         scan_line_to_use = scan_line_2_0;
         DEBUGMSGTL(("mibII/interfaces",
-                    "using linux 2.0 kernel /proc/net/dev\n"));
+                    "using linux 2.0 kernel /tmp/proc/net/dev\n"));
     }
 
 
@@ -1596,12 +1596,12 @@
 
         if (!*ifstart || ((stats = strrchr(ifstart, ':')) == NULL)) {
             snmp_log(LOG_ERR,
-                     "/proc/net/dev data format error, line ==|%s|", line);
+                     "/tmp/proc/net/dev data format error, line ==|%s|", line);
             continue;
         }
         if ((scan_line_to_use == scan_line_2_2) && ((stats - line) < 6)) {
             snmp_log(LOG_ERR,
-                     "/proc/net/dev data format error, line ==|%s|", line);
+                     "/tmp/proc/net/dev data format error, line ==|%s|", line);
         }
 
         *stats   = 0;
@@ -1621,7 +1621,7 @@
             if ((scan_line_to_use == scan_line_2_2)
                 && !strstr(line, "No statistics available"))
                 snmp_log(LOG_ERR,
-                         "/proc/net/dev data format error, line ==|%s|",
+                         "/tmp/proc/net/dev data format error, line ==|%s|",
                          line);
             continue;
         }
diff -ENwbur a/agent/mibgroup/mibII/ipv6.c b/agent/mibgroup/mibII/ipv6.c
--- a/agent/mibgroup/mibII/ipv6.c	2019-05-26 07:34:24.641629501 -0300
+++ b/agent/mibgroup/mibII/ipv6.c	2019-05-26 07:34:40.289345053 -0300
@@ -2556,7 +2556,7 @@
   free(ifni);					\
 }
 
-#define LINUX_PROC_NET_IFINET6 "/proc/net/if_inet6"
+#define LINUX_PROC_NET_IFINET6 "/tmp/proc/net/if_inet6"
 struct if_nameindex *
 linux_if_nameindex(void)
 {
